astropy/uncertainty/tests/test_containers.py

```diff
@@ -93,3 +111,119 @@ def test_earthlocation_geodetic_distribution(self):
+
+        return np.all(result & components_allclose(rep1, rep2))
+
+    def test_cartesian(self):
+        dcart = self.dsph.to_cartesian()
+        cart = self.sph.to_cartesian()
+        assert isinstance(dcart.x, Distribution)
+        assert_array_equal(dcart.x.distribution, cart.x)
+        assert_array_equal(dcart.y.distribution, cart.y)
+        assert_array_equal(dcart.z.distribution, cart.z)
+
+    def test_cartesian_roundtrip(self):
+        roundtrip = SphericalRepresentation.from_cartesian(self.dsph.to_cartesian())
+        self.assert_representation_allclose(self.get_distribution(roundtrip), self.sph)
+
+    @pytest.mark.parametrize("rep_cls", REPRESENTATION_CLASSES.values())
+    def test_other_reps(self, rep_cls):
+        drep = self.dsph.represent_as(rep_cls)
+        rep = self.sph.represent_as(rep_cls)
+        self.assert_representation_equal(self.get_distribution(drep), rep)
+
+
+class TestRepresentationWithDifferential(TestRepresentation):
+    @classmethod
+    def setup_class(cls):
+        super().setup_class()
+        cls.d_lon = Distribution(np.ones(10) << u.deg / u.hour)
+        cls.d_lat = np.ones(cls.lat.shape) << u.deg / u.hour
+        cls.d_r = 1 * u.m / u.s
+        cls.d_sph = SphericalDifferential(
+            cls.d_lon.distribution, cls.d_lat[:, np.newaxis], cls.d_r
+        )
+        cls.dd_sph = SphericalDifferential(cls.d_lon, cls.d_lat, cls.d_r)
+        cls.sph = cls.sph.with_differentials({"s": cls.d_sph})
+        cls.dsph = cls.dsph.with_differentials({"s": cls.dd_sph})
+
+    def test_cartesian(self):
+        dcart = self.dsph.represent_as(CartesianRepresentation, CartesianDifferential)
+        cart = self.sph.represent_as(CartesianRepresentation, CartesianDifferential)
+        assert isinstance(dcart.x, Distribution)
+        assert_array_equal(dcart.x.distribution, cart.x)
+        assert_array_equal(dcart.y.distribution, cart.y)
+        assert_array_equal(dcart.z.distribution, cart.z)
+        assert "s" in dcart._differentials
+        dd_cart = dcart._differentials["s"]
+        d_cart = cart._differentials["s"]
+        assert_array_equal(dd_cart.d_x.distribution, d_cart.d_x)
+        assert_array_equal(dd_cart.d_y.distribution, d_cart.d_y)
+        assert_array_equal(dd_cart.d_z.distribution, d_cart.d_z)
+
+    def test_cartesian_roundtrip(self):
+        roundtrip = self.dsph.represent_as(
+            CartesianRepresentation, CartesianDifferential
+        ).represent_as(SphericalRepresentation, SphericalDifferential)
+        self.assert_representation_allclose(self.get_distribution(roundtrip), self.sph)
+
+    @pytest.mark.parametrize("diff_cls", DIFFERENTIAL_CLASSES.values())
+    def test_other_reps(self, diff_cls):
+        repr_cls = diff_cls.base_representation
+        drep = self.dsph.represent_as(repr_cls, diff_cls)
+        rep = self.sph.represent_as(repr_cls, diff_cls)
+        self.assert_representation_equal(self.get_distribution(drep), rep)
+
+
+class TestSkyCoord:
```

Comment:
Should there be a set of tests for CoordinateFrame?

Reply:
The tests for `SkyCoord` are really limited - one should really add, say, a distribution in temperature for `AltAz`. If you prefer, I can remove the `SkyCoord` tests so we just focus on the representations (I kept it in a separate commit partially for that reason; I was just very happy it worked at all!!).

Reply:
I'm definitely happy it works as well. I was just saying that the tests should probably mimic the object hierarchy and have CoordinateFrame tests between Representation and SkyCoord tests.

Reply:
Agreed in principle,  but I think adding another skeleton test does not help that much. And full tests should be a different PR. 

Probably this all argues I should just remove the commit... Though it is sort-of nice to have at least a minimal guard against messing up SkyCoord...

Reply:
It's fine to keep.